deps/rabbit/src/rabbit_quorum_queue.erl

```diff
@@ -286,6 +278,36 @@ start_cluster(Q) ->
             Ex
     end.
 
+start_cluster(Q) ->
+    QName = amqqueue:get_name(Q),
+    Opts = amqqueue:get_options(Q),
+    Arguments = amqqueue:get_arguments(Q),
+    ActingUser = maps:get(user, Opts, ?UNKNOWN_USER),
+    QuorumSize = get_default_quorum_initial_group_size(Arguments),
+    {LeaderNode, FollowerNodes} =
+        rabbit_queue_location:select_leader_and_followers(Q, QuorumSize),
+    RaName = case qname_to_internal_name(QName) of
+                 {ok, A} ->
+                     A;
+                 {error, {too_long, N}} ->
+                     rabbit_data_coercion:to_atom(ra:new_uid(N))
+             end,
+    rabbit_log:debug("Will start up to ~w replicas for quorum ~ts with leader on node '~ts'",
+                     [QuorumSize, rabbit_misc:rs(QName), LeaderNode]),
+    case internal_declare(Q, RaName, LeaderNode, FollowerNodes) of
+        {error, cluster_not_formed, Queue} ->
+            rabbit_amqqueue:internal_delete(Queue, ActingUser),
+            AlternativeRaName = rabbit_data_coercion:to_atom(ra:new_uid(name_concat(QName))),
+            rabbit_log:debug("Could not form cluster ~ts, trying alternative name ~ts",
+                     [RaName, AlternativeRaName]),
+            case internal_declare(Q, AlternativeRaName, LeaderNode, FollowerNodes) of
```

Comment:
It would have been more elegant for `internal_declare` to handle the retry, but I understand why you dont so I think its fine.